File: android/app/src/main/kotlin/com/bockmedia/console/domain/model/SearchBrowseSessionCache.kt
package com.bockmedia.console.domain.model

/** In-memory search browse feed so the Search tab paints instantly on revisit. */
object SearchBrowseSessionCache {
    private const val TTL_MS = 600_000L

    @Volatile
    private var feed: SearchBrowseFeed? = null

    @Volatile
    private var cachedAtMs: Long = 0L

    fun peek(): SearchBrowseFeed? = feed

    fun getIfFresh(): SearchBrowseFeed? {
        val cached = feed ?: return null
        if (System.currentTimeMillis() - cachedAtMs > TTL_MS) return null
        return cached
    }

    fun put(value: SearchBrowseFeed) {
        if (value.pickedForYou.isEmpty() && value.newReleases.isEmpty() && value.genres.isEmpty()) return
        feed = value
        cachedAtMs = System.currentTimeMillis()
    }

    fun invalidate() {
        feed = null
        cachedAtMs = 0L
    }
}

File: ios/BockMedia/Domain/SearchBrowseSessionCache.swift
import Foundation

enum SearchBrowseSessionCache {
    private static let ttlMs: Int64 = 600_000
    private static var cachedFeed: SearchBrowseFeed?
    private static var cachedAtMs: Int64 = 0

    static func peek() -> SearchBrowseFeed? { cachedFeed }

    static func getIfFresh() -> SearchBrowseFeed? {
        guard let cachedFeed else { return nil }
        guard Int64(Date().timeIntervalSince1970 * 1000) - cachedAtMs <= ttlMs else { return nil }
        return cachedFeed
    }

    static func put(_ feed: SearchBrowseFeed) {
        guard !feed.pickedForYou.isEmpty || !feed.newReleases.isEmpty || !feed.genres.isEmpty else { return }
        cachedFeed = feed
        cachedAtMs = Int64(Date().timeIntervalSince1970 * 1000)
    }

    static func invalidate() {
        cachedFeed = nil
        cachedAtMs = 0
    }
}
